Framework rewrite documentation (#45)
wintercms/docs: console-asset-compilation.md

<a name="requirements"></a>
### Requirements

To take advantage of Mix asset compilation, you must have Node and the Node package manager (NPM) installed in your development environment. This will be dependent on your operating system - please review the [Download NodeJS](https://nodejs.org/en/download/) page for more information on installing Node.

[Laravel Mix](https://laravel-mix.com/) should also be present in the `package.json` file for any packages that will be using it (either in `dependencies` or a `devDependencies`) but if it is not specified in the project's `package.json` file then it can be optionally automatically added when running the [`mix:install`](#mix-install) command.

<a name="registering-packages"></a>
## Registering a package

Registering for asset compilation through Mix is very easy. Automatic registration should meet your needs most of the time, and if not there are several methods available to manually register Mix packages.

<a name="automatic-registration"></a>
### Automatic registration

> By default, Winter will scan all available and enabled Modules, Plugins, & Themes for the presence of a `winter.mix.js` file under each extension's root folder (i.e. `modules/system/winter.mix.js`, `plugins/myauthor/myplugin/winter.mix.js`, or `themes/mytheme/winter.mix.js`).

Author: ```suggestion
By default, Winter will scan all available and enabled modules, plugins and themes for the presence of a `winter.mix.js` file under each extension's root folder (i.e. `modules/system/winter.mix.js`, `plugins/myauthor/myplugin/winter.mix.js`, or `themes/mytheme/winter.mix.js`).
```

> If the `winter.mix.js` file is found it will be automatically registered as a package and will show up when running the Mix commands. Most of the time this should be all you need to do in order to get started with Laravel Mix based asset compilation in Winter CMS.

Author: ```suggestion
If the `winter.mix.js` file is found, it will be automatically registered as a package with an automatically generated package name, and will show up when running the Mix commands. Most of the time, this should be all you need to do in order to get started with Laravel Mix asset compilation in Winter CMS.
```

<a name="registering-plugins"></a>
### Registering plugins

To register frontend assets to be compiled through Mix in your plugin, simply return an array with the package names as the keys and the package paths relative to the plugin's directory as the values to register from your [`Plugin.php`](../plugin/registration) registration file's `registerMixPackages()` method. See below example.

```php
public function registerMixPackages()
{
    return [

>         'package-name-2' => 'assets/js/build.js',
> 
>         // winter.mix.js is assumed to be the config file in this path
>         'package-name-3' => 'assets/css',

Reviewer: ```suggestion
        'custom-package-name' => 'assets/js/build.mix.js',
```
